### DISCUSSION TITLE
Depending on the value passed to the family, it is recognized as a different provider.

### DISCUSSION THREAD

**Discussion by nero-angela** [rrousselGit/riverpod#2128] [Q&A]
Hi I'm studying `family`, but I don't understand how it works.

At first, after seeing [this discussion](https://github.com/rrousselGit/riverpod/discussions/861), I thought the family passed when using `ref.read()` was a meaningless value. But looking at the following [example code](https://dartpad.dev/?id=ad9fd181957d4aba0def0e889ff39da7), it seems that different providers are recognized depending on the values passed by `ref.read(counterProvider(10))` and `ref.read(counterProvider(1))`. Is this the intended function??

<img width="1154" alt="image" src="https://user-images.githubusercontent.com/26322627/216495887-63d61271-0c1c-4305-ac3e-623f6f9e3139.png">

**Comment by rrousselGit** [comment]
Yes. That's the whole point

The parameter isn't a "parameter for initialization".
It's a unique key pointing to a different state.


View `YourProvider.family<Parameter, Result>` as a `Map<Parameter, YourProvider<Result>>`

**Reply by nero-angela** [reply]
Thank you for answer.
Is there any way to pass the initial parameters without the unique key function?

**Reply by rrousselGit** [reply]
No.
